src/index.ts

```diff
@@ -272,6 +273,8 @@ export interface SetupOptions {
 	onLogin?: (
 		user: Pick<DbUser, typeof defaultFindUser$select[number]>,
 	) => PromiseLike<void> | void;
+
+	getNewUserRole?: (user: AnyObject) => string;
```

Comment:
As per `GetUrlFunction` and `SetupFunction` above, let's give the callback a name that can be imported by clients:

```ts
  // src/infra/auth/auth.ts
  export type GetNewUserRoleFunction = (user: AnyObject) => string;
  
  // this file:
  getNewUserRole?: GetNewUserRoleFunction;
```